docs/xks/developer-guide/secrets-management.md

```diff
@@ -377,3 +377,52 @@ spec:
             volumeAttributes:
               secretProviderClass: foo
 ```
+
+## Troubleshooting
+
+There are a lot of things that can go wrong when configuring Secrets. Here are some pointers for things to check:
+
+### Did you forget to declare the `SecretProviderClass`?
+
+Remember, getting access to your secrets consists of 2 separate parts:
+
+- a `SecretProviderClass`, which tells Kubernetes where it can get the stored secret
+- a mount of the provided `secrets-store` for your deployment
+
+### Verifying your loaded YAML
+
+By running `kubectl get secretproviderclasspodstatuses -o yaml` you can get a lot of information about if and how your secrets got correctly loaded. Check here first!
+For example, look out to see that all the secrets you expect to see are available, and that they are mounted:
+
+```yaml
+status:
+  mounted: true
+```
+
+### Is your key vault correctly configured, and does the pod have access to it?
+
+Speaking from experience, it is all too easy to setup access to the wrong key vault. If you are accessing the right key vault and you are using Azure, double check that
+`usePodIdentity: "true"` is set on the `SecretProviderClass` and it has the `aadpodidbinding` label.
```

Comment:
Please add a note about "make sure that the labels is defined in the pod template and not only in the deployment label."
Or something similar.